src/posix/platform/netif.cpp

```diff
@@ -1339,7 +1355,14 @@ static void processNetifAddrEvent(otInstance *aInstance, struct nlmsghdr *aNetli
                     netAddr.mAddress      = addr;
                     netAddr.mPrefixLength = ifaddr->ifa_prefixlen;
 
-                    error = otIp6AddUnicastAddress(aInstance, &netAddr);
+                    if (otIp6HasUnicastAddress(aInstance, &netAddr.mAddress))
+                    {
+                        error = OT_ERROR_NONE;
+                    }
+                    else
+                    {
+                        error = otIp6AddUnicastAddress(aInstance, &netAddr);
+                    }
```

Comment:
I recommend calling `otIp6AddUnicastAddress()`, even if `otIp6HasUnicastAddress()` returns true. This ensures that we update the preferred/valid flags in case they've changed.

The `otIp6AddUnicastAddress()` function should handle the scenario where the same address was added previously and return `OT_ERROR_NONE`. It will only fail and return an error from the `IsUnicastAddressExternal()` check if the address is one that the Thread stack itself adds (like RLOC address, mesh-local EID, etc.).

https://github.com/openthread/openthread/blob/32f462ff34d73d8011d349387f1433d1b9305117/src/core/net/netif.cpp#L437-L472

Reply:
Looks reasonable. Updated.